### ISSUE TITLE
Plugin Marketplace - Simple sites show 'Untested with your version of WordPress' for all plugins

### ISSUE THREAD

**Issue by Greatdane** [Automattic/wp-calypso#60323]
### Quick summary

On the calypsofied plugin marketplace (https://wordpress.com/plugins/site_url), if you visit this on a simple site, nearly all plugins say `Untested with your version of WordPress`. I assume this is because a simple site has an unrecognized WordPress version?

It makes it look like WordPress.com is using an outdated version on WordPress.

Image of a Simple site (all plans, including Business+);

<img width="1090" alt="Markup 2022-01-20 at 17 03 06" src="https://user-images.githubusercontent.com/5075840/150440324-c7a96b52-efa9-4265-9f62-3dfac567ef3a.png">

Image of an Atomic site;

<img width="1111" alt="Markup 2022-01-21 at 08 50 12" src="https://user-images.githubusercontent.com/5075840/150440526-46c7b49f-3442-4f21-81de-12766ead2cda.png">

### Steps to reproduce

1. Navigate to plugins on any Simple site, with any plan. 
2. You will see nearly all plugins (including WooCommerce and Jetpack) will show `Untested with your version of WordPress`.

### What you expected to happen

Nearly all plugins (including our own) should show as being compatible with the version on WordPress on WordPress.com 

### What actually happened

We see `Untested with your version of WordPress`.

### Simple, Atomic or both?

Simple

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

There is no user impact

**Comment by cpapazoglou** [comment]
"Untested with your version of WordPress" means that the plugin developer hasn't tested their plugin against that WordPress version. 

Since on WPCOM we are always using the latest versions (even beta, before releasing them to Atomic), there will be often cases that this message appears on Simple sites. Simple sites are now in `5.9` while Atomic sites in < `5.9`.

As you can see, there are developers who have already reported that their plugin is tested with `5.9`.
![SS 2022-01-21 at 12 09 14](https://user-images.githubusercontent.com/12430020/150508771-5a2e6f50-b36b-4189-9fc0-8dda99aefa83.png)

So, to recap, the message stands true. It's not a bug. But, I can understand that this might intimidate users. FWIW, it would also display in WP Admin `/wp-admin/plugin-install.php` page.

![SS 2022-01-21 at 12 16 55](https://user-images.githubusercontent.com/12430020/150509801-e654265d-b62a-4ece-bc2b-ea702b9d39e0.png)


@SaxonF, @vajrasar, do you think we should _avoid_ showing that message in the plugins list item component? BTW, if we do so, there might be valuable info missing from the user

**Comment by cpapazoglou** [comment]
One idea would be checking against the same WordPress version as in Atomic, @WBerredo is that something we can do? Or avoid checking beta / rc versions.

**Comment by WBerredo** [comment]
The version we use on atomic currently is `5.8.3`. While the version is used on simple sites is `5.9-beta4-52009`.
So even if we remove the beta check we would still be showing less coverage in free sites than atomic sites.

There is a possibility to know which version atomic is using? It would be more trustworthy info.

**Comment by SaxonF** [comment]
One option is to move that message to the plug-in detail page closely mapped to the install button. 

Also, for plugins that we feature or are in the popular list I wonder if it's worthwhile creating a process around which we test those internally?

**Comment by cpapazoglou** [comment]
@SaxonF there is a paradox happening here:
- Simple sites are on `5.9-beta4-52009` and most plugins are **untested** 
- Atomic sites are on `5.8.3` and most plugins are **tested** 

Trying to install a plugin from as a simple site will transfer the site to Atomic, in which case the plugin would be tested. So, what we should really do is check against the version of WordPress in which the plugin will be installed (in this case Atomic, `5.8.3`).

@WBerredo 5.8.3 is the previous version to 5.9-beta* http://api.wordpress.org/core/stable-check/1.0/

I don't think there is an API retrieving the current Atomic version. @SaxonF how would you feel hiding that kind of info from Simple sites? Unexperienced users would be intimidated by this messaging but actually the plugin will work in most cases (even if it's not maintained for years).

**Comment by SaxonF** [comment]
That sounds fine to me but I also lack a lot of historical context so might not be the best person to ask in this case.

I'm sure we'd find out pretty quick if we need to surface more info via feedback, error logs pertaining to plugins etc

**Comment by cpapazoglou** [comment]
Discussed it in pod call, let's remove that kind of information from Simple site users.
